### PR TITLE
Add "Copy story data" button to error boundary screen 

### LINKED ISSUE #11394: Add "Copy story data" button to error boundary screen
## Feature Description

It would be useful to add an option for web stories users to export a story as a JSON file to provide to support for debugging. 

This would simply be an export function and would not allow users to export/import story data. This would allow us in the support forums to request this data from the user to help with our troubleshooting of their story issues.

The feature would work much like other plugins with an export settings feature.  

## Alternatives Considered

1. This could be included in the Site Health Information rather than adding an option for the user to download a JSON file. 

2. An alternative to this would be to continue using the process we have in place which is less efficient but does work. This improvement would eliminate many steps in the process making everything much more efficient. 

## Additional Context

N/A

### PR DESCRIPTION
**Pull request by timarney**
## Context

It would be useful to add an option for web stories users to export a story as a JSON file to provide to support for debugging.

This would simply be an export function and would not allow users to export/import story data. This would allow us in the support forums to request this data from the user to help with our troubleshooting of their story issues.

## Summary

Adds a `copy story data` to Error Boundary screen

See Video:
https://www.youtube.com/watch?v=AKZv9o0-F0w

Screen Shot 2022-10-27 at 8 19 43 AM

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.  To test I needed to modify the code to throw an error i.e. a non existent callback or you can modify the error state of the ErrorBoundry

```
 state = {
    error: null,
    errorInfo: null,
  };
```

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11394

### CODE REVIEW

File: packages/story-editor/src/components/errorBoundary/copyStoryDataToClipboard.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  const { pages, current, selection, story } = reducerState;
+
+  const copyToClipboard = useCallback(async () => {
+    const jsonStr = JSON.stringify(
+      { pages, current, selection, story },
+      null,
+      2
+    );
+    await navigator.clipboard.writeText(jsonStr);
+    alert(__('Copied to clipboard', 'web-stories'));
```
Comment (spacedmonkey) [inline packages/story-editor/src/components/errorBoundary/copyStoryDataToClipboard.js]: Wrap in a try / catch, just incase.

**Review by spacedmonkey — APPROVED** [review]
Works well in my testing.

File: packages/story-editor/src/components/errorBoundary/index.js [hunk 2]
```diff
@@ -46,15 +47,16 @@ class ErrorBoundary extends Component {
...
     return this.props.children;
   }
 }
 
-const shouldDisableErrorBoundaries =
-  typeof WEB_STORIES_DISABLE_ERROR_BOUNDARIES !== 'undefined' &&
-  WEB_STORIES_DISABLE_ERROR_BOUNDARIES === 'true';
-export default shouldDisableErrorBoundaries
-  ? ({ children }) => children
-  : ErrorBoundary;
```
Comment (swissspidy) [inline packages/story-editor/src/components/errorBoundary/index.js]: Why was this `WEB_STORIES_DISABLE_ERROR_BOUNDARIES` check removed?

If we wanna remove it, we should do so in a separate PR since this is referenced in many places. Otherwise we end up with dead code.

File: packages/story-editor/src/components/errorBoundary/copyStoryDataToClipboard.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ * Internal dependencies
+ */
+import { useStory } from '../../app/story';
+
+function CopyStoryDataToClipboard() {
+  const { reducerState } = useStory(({ internal }) => ({
+    reducerState: internal.reducerState,
+  }));
+
+  const { pages, current, selection, story } = reducerState;
```
Comment (swissspidy) [inline packages/story-editor/src/components/errorBoundary/copyStoryDataToClipboard.js]:
```suggestion
  const { pages, current, selection, story  } = useStory(({ internal }) => ({
    pages: internal.reducerState.pages,
    current: internal.reducerState.current,
    selection: internal.reducerState.selection,
    story: internal.reducerState.story,
  }));
```

Generally, we'd want to reduce the number of re-renders, which we can achieve by only returning what we need.

Example: If `reducer.foo` constantly changes, with the current version this component would always re-render. But with this suggestion it won't, because it's not actually used.

File: packages/story-editor/src/components/errorBoundary/copyStoryDataToClipboard.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+ * External dependencies
+ */
+import { __ } from '@googleforcreators/i18n';
+import {
+  Button,
+  BUTTON_SIZES,
+  BUTTON_TYPES,
+  BUTTON_VARIANTS,
+} from '@googleforcreators/design-system';
+import { useCallback } from 'react';
```
Comment (swissspidy) [inline packages/story-editor/src/components/errorBoundary/copyStoryDataToClipboard.js]: We import from `@googleforcreators/react`

File: packages/story-editor/src/components/errorBoundary/errorActions.js [hunk 5]
```diff
@@ -86,6 +91,7 @@ function ErrorActions({ error, errorInfo }) {
         >
           {__('Reload', 'web-stories')}
         </Button>
+        {withStoryData && <CopyStoryDataToClipboard />}
```
Comment (swissspidy) [inline packages/story-editor/src/components/errorBoundary/errorActions.js]: Do we actually need this check?

`CopyStoryDataToClipboard` can just `return null` if there's no story data available.

File: packages/story-editor/src/components/errorBoundary/index.js (hunk 2 above)
Comment (timarney) [reply packages/story-editor/src/components/errorBoundary/index.js]: Bad commit on my end --- at one point I created a new error boundary file.  Reverted now.

File: packages/story-editor/src/components/errorBoundary/errorActions.js (hunk 5 above)
Comment (timarney) [reply packages/story-editor/src/components/errorBoundary/errorActions.js]: This is a contrived example but if something doesn't catch higher up I think it will just error out like this. 

Screen Shot 2022-10-31 at 7 40 04 AM

In this case I forced a miss  ☝️ of the higher up boundries.

Thinking it's better to be specific vs potentially have it error out trying to get data isn't available to access.  Thoughts?

Comment (swissspidy) [reply packages/story-editor/src/components/errorBoundary/errorActions.js]: My concern is that this is error prone when code gets moved around, and that we can just make `CopyStoryDataToClipboard` more robust instead of adding these`withStoryData` checks everywhere.

Something like:

```js
  // Get story data if it's available, otherwise return early.
  const { pages, current, selection, story } = useStory(({ internal }) => ({
    pages: internal?.reducerState?.pages,
    current: internal?.reducerState?.current,
    selection: internal?.reducerState?.selection,
    story: internal?.reducerState?.story,
  }));

  if (!pages || !current || !selection || !story) {
    return null;
  }
```

WDYT?

Comment (timarney) [reply packages/story-editor/src/components/errorBoundary/errorActions.js]: Okay updated --- I don't think the example I showed can happen give the way the boundaries are setup,
